tools/HubMigrator/HubMigrator.dm

```diff
@@ -0,0 +1,166 @@
+//Misc Medal hub IDs
+#define MEDAL_METEOR 			"Your Life Before Your Eyes"
+#define MEDAL_PULSE 			"Jackpot"
+#define MEDAL_TIMEWASTE 		"Overextended The Joke"
+#define MEDAL_RODSUPLEX 		"Feat of Strength"
+#define MEDAL_CLOWNCARKING 		"Round and Full"
+#define MEDAL_THANKSALOT 		"The Best Driver"
+#define MEDAL_HELBITALJANKEN	"Hel-bent on Winning"
+#define MEDAL_MATERIALCRAFT 	"Getting an Upgrade"
+
+
+//Boss medals
+
+// Medal hub IDs for boss medals (Pre-fixes)
+#define BOSS_MEDAL_ANY		  "Boss Killer"
+#define BOSS_MEDAL_MINER	  "Blood-drunk Miner Killer"
+#define BOSS_MEDAL_BUBBLEGUM  "Bubblegum Killer"
+#define BOSS_MEDAL_COLOSSUS	  "Colossus Killer"
+#define BOSS_MEDAL_DRAKE	  "Drake Killer"
+#define BOSS_MEDAL_HIEROPHANT "Hierophant Killer"
+#define BOSS_MEDAL_LEGION	  "Legion Killer"
+#define BOSS_MEDAL_TENDRIL	  "Tendril Exterminator"
+#define BOSS_MEDAL_SWARMERS   "Swarmer Beacon Killer"
+
+#define BOSS_MEDAL_MINER_CRUSHER	  	"Blood-drunk Miner Crusher"
+#define BOSS_MEDAL_BUBBLEGUM_CRUSHER  	"Bubblegum Crusher"
+#define BOSS_MEDAL_COLOSSUS_CRUSHER	  	"Colossus Crusher"
+#define BOSS_MEDAL_DRAKE_CRUSHER	  	"Drake Crusher"
+#define BOSS_MEDAL_HIEROPHANT_CRUSHER 	"Hierophant Crusher"
+#define BOSS_MEDAL_LEGION_CRUSHER	 	"Legion Crusher"
+#define BOSS_MEDAL_SWARMERS_CRUSHER		"Swarmer Beacon Crusher"
+
+// Medal hub IDs for boss-kill scores
+#define BOSS_SCORE 	         "Bosses Killed"
+#define MINER_SCORE 		 "BDMs Killed"
+#define BUBBLEGUM_SCORE 	 "Bubblegum Killed"
+#define COLOSSUS_SCORE 	     "Colossus Killed"
+#define DRAKE_SCORE 	     "Drakes Killed"
+#define HIEROPHANT_SCORE 	 "Hierophants Killed"
+#define LEGION_SCORE 	     "Legion Killed"
+#define SWARMER_BEACON_SCORE "Swarmer Beacs Killed"
+#define TENDRIL_CLEAR_SCORE	 "Tendrils Killed"
+
+
+
+//Migration script generation
+//Replace hub information and fire to generate hub_migration.sql script to use.
+/mob/verb/generate_migration_script()
```

Comment:
I think this'd be better as an external script. I've used python for parsing byond.com before and it handled the volume and inserting to DB oodles better than byond did https://github.com/tgstation/tgstation/blob/master/SQL/populate_key_2018-07-09.py

On other hand I won't blame you if you don't want to rewrite this, it looks like a hassle.

Reply:
nevermind thanks byond api